File: src/main/java/baboon/industry/recipe/RecipesCannery.java
package baboon.industry.recipe;

import baboon.industry.item.IndustryItems;
import net.minecraft.core.item.Item;
import net.minecraft.core.item.ItemStack;

import java.util.HashMap;

public class RecipesCannery {
    private static final HashMap<Integer, ItemStack> recipeList = new HashMap<>();

    public RecipesCannery() {
        addRecipe(Item.foodFishRaw.id, 1);
        addRecipe(Item.foodCookie.id, 1);
        addRecipe(Item.cherry.id, 1);
        addRecipe(Item.foodApple.id, 2);
        addRecipe(Item.foodPorkchopRaw.id, 2);
        addRecipe(Item.foodBread.id, 3);
        addRecipe(Item.foodFishCooked.id, 3);
        addRecipe(Item.foodPorkchopCooked.id, 4);
        addRecipe(Item.foodStewMushroom.id, 5);
        addRecipe(Item.foodCake.id, 6);
        addRecipe(Item.foodAppleGold.id, 21);

        addSpecialRecipe(IndustryItems.ingotUranium.id, new ItemStack(IndustryItems.cellUranium, 1));
    }

    public static void addRecipe(int input, int outputCount) {
        recipeList.put(input, new ItemStack(IndustryItems.canFood, outputCount));
    }

    public static void addSpecialRecipe(int input, ItemStack output) {
        recipeList.put(input, output);
    }

    public ItemStack getRecipeResult(int i) {
        return recipeList.get(i);
    }

    public HashMap<Integer, ItemStack> getRecipeList() {
        return recipeList;
    }
}

File: src/main/kotlin/turniplabs/industry/recipes/RecipesCannery.kt
package turniplabs.industry.recipes

import net.minecraft.core.item.Item
import net.minecraft.core.item.ItemStack
import turniplabs.industry.items.IndustryItems

object RecipesCannery {
    private val recipeList = HashMap<Any?, Any?>()

    init {
        addRecipe(Item.foodCookie, 1)
        addRecipe(Item.foodFishRaw, 1)
        addRecipe(Item.cherry, 1)
        addRecipe(Item.foodApple, 2)
        addRecipe(Item.foodPorkchopRaw, 3)
        addRecipe(Item.foodBread, 3)
        addRecipe(Item.foodFishCooked, 3)
        addRecipe(Item.foodPorkchopCooked, 4)
        addRecipe(Item.bucketIcecream, 5)
        addRecipe(Item.foodStewMushroom, 5)
        addRecipe(Item.foodCake, 7)
        addRecipe(Item.foodAppleGold, 21)

        addSpecialRecipe(IndustryItems.uraniumIngot, IndustryItems.cellUranium, 1)
    }

    fun addRecipe(input: Item, canOutputCount: Int) {
        recipeList[input.id] = ItemStack(IndustryItems.canFull, canOutputCount)
    }

    fun addSpecialRecipe(input: Item, outputItem: Item, outputCount: Int) {
        recipeList[input.id] = ItemStack(outputItem, outputCount)
    }

    fun getResult(i: Int): ItemStack? {
        return recipeList[i] as ItemStack?
    }

    fun getRecipeList(): HashMap<Any?, Any?> {
        return recipeList
    }
}